include/Token/TokenWrapper.h

```diff
@@ -42,14 +42,34 @@ class TokenWrapper {
   std::vector<Token *> extractToken(llvm::GlobalVariable *);
   std::vector<Token *> extractToken(llvm::CallInst *);
   std::vector<Token *> extractToken(llvm::Argument *, llvm::Function *);
+  std::vector<Token *> extractPHINodeToken(llvm::Instruction *);
 
   template <typename Ty> std::pair<int, int> extractStatementType(Ty *);
-
+  
+  std::vector<int> extractPHINodeStatementType(llvm::Instruction *);
   template <typename GEP> Token *handleGEPUtil(GEP *, Token *);
 
   ~TokenWrapper();
 };
 
 } // namespace spatial
 
+namespace spatial{
+  /// extractStatementType - Returns the relative level of redirection based of
+  /// LHS and RHS on the statement
+	template <typename Ty>
+  std::pair<int, int> TokenWrapper::extractStatementType(Ty *Inst) {
```

Comment:
This logic has now moved to InstModel/GenericInstModel/GenericInstModel.cpp, you may want to do your changes there